File: app/src/main/java/net/twisterrob/android/view/color/swatches/pixel/color/LinearHueGradient.java
package net.twisterrob.android.view.color.swatches.pixel.color;

import net.twisterrob.android.view.color.ColorMath;

public class LinearHueGradient implements PixelColor {
    private int w;

    @Override
    public void initializeInvariants(int w, int h) {
        this.w = w;
    }

    @Override
    public int getPixelColorAt(int x, int y) {
        return ColorMath.fromHsb((float) x / w, 1, 1, 1);
    }
}

File: feature/base/src/main/kotlin/net/twisterrob/android/view/color/swatches/pixel/color/LinearHueGradient.kt
package net.twisterrob.android.view.color.swatches.pixel.color

import androidx.annotation.ColorInt
import androidx.annotation.IntRange
import net.twisterrob.android.view.color.fromHsb

class LinearHueGradient : PixelColor {

	@IntRange(from = 0)
	private var w: Int = 0

	override fun initializeInvariants(@IntRange(from = 0) w: Int, @IntRange(from = 0) h: Int) {
		this.w = w
	}

	@ColorInt
	override fun getPixelColorAt(@IntRange(from = 0/*, to = w*/) x: Int, @IntRange(from = 0/*, to = h*/) y: Int): Int {
		return fromHsb(x.toFloat() / w.toFloat(), 1.0f, 1.0f, 1.0f)
	}
}
